flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/AbstractRestHandler.java

```diff
@@ -117,11 +117,26 @@ protected void respondAsLeader(final ChannelHandlerContext ctx, Routed routed, T
 				}
 			}
 
+			final HandlerRequest<R, M> handlerRequest;
+
+			try {
+				handlerRequest = new HandlerRequest<>(request, messageHeaders.getUnresolvedMessageParameters(), routed.pathParams(), routed.queryParams());
+			} catch (HandlerRequestException hre) {
+				log.error("Could not create the handler request.", hre);
+
+				HandlerUtils.sendErrorResponse(
+					ctx,
+					httpRequest,
+					new ErrorResponseBody(String.format("Could not create the handler request: %s", hre.getMessage())),
```

Comment:
HandlerRequests is an external concept that we imo should not expose. I would just say "Bad request, could not parse parameters."

Reply:
Sure, can change this.